### PR TITLE
Use new litserve accelerator="auto" setting

### PR DESCRIPTION
**Pull request by rasbt**
Updates the hardcoded "cuda" to "auto"

### CODE REVIEW

File: litgpt/deploy/serve.py [hunk 1]
```diff
@@ -114,7 +114,8 @@ def run_server(
             generated text but can also lead to more incoherent texts.
         max_new_tokens: The number of generation steps to take.
         devices: How many devices/GPUs to use.
-        accelerator: The type of accelerator to use. For example, "cuda" or "cpu".
+        accelerator: The type of accelerator to use. For example, "auto", "cuda", "cpu", or "mps".capitalize
```
Comment (rasbt) [inline litgpt/deploy/serve.py:117]:
```suggestion
        accelerator: The type of accelerator to use. For example, "auto", "cuda", "cpu", or "mps".
```

**Comment by rasbt** [comment]
CC @lantiga & @aniketmaurya for code review

File: litgpt/deploy/serve.py [hunk 2]
```diff
@@ -114,7 +114,8 @@ def run_server(
             generated text but can also lead to more incoherent texts.
         max_new_tokens: The number of generation steps to take.
         devices: How many devices/GPUs to use.
-        accelerator: The type of accelerator to use. For example, "cuda" or "cpu".
+        accelerator: The type of accelerator to use. For example, "auto", "cuda", "cpu", or "mps".
```
Comment (carmocca) [inline litgpt/deploy/serve.py:117]: You should update the TODO in L43 too

File: pyproject.toml [hunk 3]
```diff
@@ -12,7 +12,7 @@ dependencies = [
     "torch>=2.2.0",
     "lightning==2.3.0.dev20240328",
     "jsonargparse[signatures]>=4.27.6",
-    "litserve==0.0.0.dev2",   # imported by litgpt.deploy
+    "litserve @ git+https://github.com/Lightning-AI/litserve.git@aefcb13721b97b9da4ed9f7166b2f870bedfd92f", # imported by litgpt.deploy
```
Comment (carmocca) [inline pyproject.toml:15]: We cannot release litgpt with a git pin, there needs to be a litserve release first

**Review by aniketmaurya — APPROVED** [review]
LGTM 🚀

File: pyproject.toml (hunk 3 above)
Comment (lantiga) [reply pyproject.toml:15]: I’ll release 0.1.0 as soon as I have a decent connection

**Pull request merged**